pkg/config/base_path.go

```diff
@@ -0,0 +1,138 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/charmbracelet/log"
+	"github.com/cloudposse/atmos/pkg/schema"
+	u "github.com/cloudposse/atmos/pkg/utils"
+)
+
+// Derive the effective base-base based on multiple inferences
+func (cl *ConfigLoader) BasePathComputing(configAndStacksInfo schema.ConfigAndStacksInfo) (string, error) {
+	// Check base path from CLI argument
+	if configAndStacksInfo.BasePathFromArg != "" {
+		return cl.resolveAndValidatePath(configAndStacksInfo.BasePathFromArg, "--config")
+	}
+
+	// Check base path from ATMOS_BASE_PATH environment variable
+	if envBasePath := os.Getenv("ATMOS_BASE_PATH"); envBasePath != "" {
+		return cl.resolveAndValidatePath(envBasePath, "ATMOS_BASE_PATH")
+	}
+
+	// Check base path from configuration
+	if cl.atmosConfig.BasePath != "" {
+		source := "atmos config"
+		if !filepath.IsAbs(cl.atmosConfig.BasePath) {
+			// If relative, make it absolute based on the current working directory
+			absPath, err := filepath.Abs(cl.atmosConfig.BasePath)
+			if err != nil {
+				return "", fmt.Errorf("failed to resolve relative base path from %s: %w", source, err)
+			}
+			cl.atmosConfig.BasePath = absPath
+		}
+		return cl.resolveAndValidatePath(cl.atmosConfig.BasePath, source)
+	}
+
+	resolvedPath, found := cl.inferBasePath()
+	if found {
+		absPath, err := filepath.Abs(resolvedPath)
+		if err != nil {
+			return "", err
+		}
+		log.Debug("base path derived from infra", "base_path", absPath)
+		return absPath, nil
+	}
+	// Set base_path to absolute path of ./
+	absPath, err := filepath.Abs("./")
+	if err != nil {
+		return "", err
+	}
+	return absPath, nil
+}
+
+// Helper to resolve and validate base path
+func (cl *ConfigLoader) resolveAndValidatePath(path string, source string) (string, error) {
```

Comment:
It's core function is to resolve the path. Validation and error handling is just something every function should do.

```suggestion
// Helper to resolve and validate base path
func (cl *ConfigLoader) resolveAbsolutePath(path string, source string) (string, error) {
```